packages/react-components/react-button/src/components/Button/useButton.ts

```diff
@@ -11,11 +10,11 @@ import type { ButtonProps, ButtonState } from './Button.types';
  */
 export const useButton_unstable = (
   props: ButtonProps,
-  ref: React.Ref<HTMLButtonElement | HTMLAnchorElement>,
+  ref: React.Ref<HTMLButtonElement | HTMLAnchorElement | HTMLDivElement>,
```

Comment:
Do we need to add `HTMLDivElement` to each one here? I don't like it because it's not in the original type.

Reply:
Well, the reason for that it's because `ARIAButtonSlotProps` now supports `div`, which mean that the `ref` inferred from using this type will result in `HTMLButtonElement | HTMLAnchorElement | HTMLDivElement`, which means that if we type it as `HTMLButtonElement | HTMLAnchorElement` we'll be getting a conflict, like this:

> Argument of type `ForwardedRef<HTMLButtonElement | HTMLAnchorElement | HTMLDivElement>` is not assignable to parameter of type `Ref<HTMLButtonElement | HTMLAnchorElement>`.

Reply:
Ok, I added a generic type that allows us to maintain `'a'` as being the only alternative for `Button` implementation. ✅
https://github.com/microsoft/fluentui/blob/28ef2eb7fc00c268f42799c27b95cb73145c12a4/packages/react-components/react-aria/src/hooks/useARIAButton.ts#L6